packages/client/hmi-client/src/workflow/ops/calibrate-julia/tera-calibrate-node-julia.vue

```diff
@@ -231,17 +233,32 @@ const runCalibrate = async () => {
 	}
 };
 
+const handlingProgress = (message: string) => {
+	const parsedMessage = JSON.parse(message);
+	console.log(parsedMessage);
+};
+
 const getStatus = async (simulationId: string) => {
 	showSpinner.value = true;
 	if (!simulationId) return;
 
 	const runIds = [simulationId];
+
+	// open a connection for each run id and handle the messages
+	runIds.forEach((id) => {
+		eventSourceManager.openConnection(id, `/simulations/${id}/sciml-intermediate-results`);
+		eventSourceManager.setMessageHandler(id, handlingProgress);
+	});
+
 	poller
 		.setInterval(3000)
 		.setThreshold(300)
 		.setPollAction(async () => simulationPollAction(runIds, props.node, progress, emit));
 	const pollerResults = await poller.start();
 
+	// closing event source connections
+	runIds.forEach((id) => eventSourceManager.closeConnection(id));
+
```

Comment:
Why are you closing them right after opening them?

Reply:
This doesn't close until the poller above finishes polling (i.e. knows if the job finished or not).